### PR TITLE
[WIP] - Add event date property to event model

### PR DESCRIPTION
**Pull request by duduribeiro**
Hi guys. I don't know if this is a valid PR for speakerfight.

With this PR, I will add a property for an event to mark when the event will occurs. This will easily help who pretends to submit a proposal to check when the event will occurs. This way, the submitter can easily check his agenda.

This will be the screen with the new field:
image

thanks a lot.

🍻

### CODE REVIEW

**Comment by luanfonceca** [comment]
Of course it is valid <3
If you can cover this with at least one test i will aprove it right away

o/

**Comment by duduribeiro** [comment]
Thanks for the feedback @luanfonceca =D.

I'm not a python/django developer. 

I will read the tests parts and add a test to cover it.

Thanks a lot

**Comment by luanfonceca** [comment]
Ok, if you need some help come join me on the http://gitter.im/luanfonceca/speakerfight

**Comment by mauricioabreu** [comment]
Hey @duduribeiro need some help here?
How can I help you to finish this amazing contribution?

**Comment by duduribeiro** [comment]
Hey @mauricioabreu I will continue work on this.

Can you guide me to the testing part?

**Comment by mauricioabreu** [comment]
@duduribeiro of course, I am glad you asked for help.

1 - You probably need to redo your migration because now we have 17 migration (without this one). Yours should be 18th. On your branch, rebase your branch with master, remove your migration and recreate it.

2 - About the test, you can simply create an event with `event_date` set, make a GET request to the URL and check if the response contains the date.
Something like:
```python
event = Event.objects.create(**event_data)
event.event_date = timezone.today()
response = self.client.get(url, args={'event_slug': event.slug})
self.assertResponseContains(response, event.event_date)
```
You can propably reuse some existent test instead of creating a new one.

**Comment by mauricioabreu** [comment]
After doing all this, could you please insert a screenshot here? 
Thank you!

**Comment by duduribeiro** [comment]
hey @mauricioabreu, I added a test and rebased it. Can you check if everything is ok? Thanks

**Comment by duduribeiro** [comment]
image

**Comment by mauricioabreu** [comment]
@luanfonceca can you review it?

**Comment by luanfonceca** [comment]
For me, is missing something to tell that this date is the event date.
Something like if we create a phrase:

"O evento: 'Teste de Evento' acontecerá no dia 30 de Setembro de 2017 e a chamada de trabalho se encerra no dia 20 de Setembro de 2017."
If we make something like that we need to cover the cases of past events and due dated events.

**Comment by duduribeiro** [comment]
I think that the long message is nice. 

But we don't need to cover past events and due dated events.
- Past events will have `null` on the event_date and will not display it in the page.
- Due dated events will not receive new proposals but I think that it is nice to keep displaying when the event will occur.

WDYT? cc @mauricioabreu

**Comment by luanfonceca** [comment]
There is a flag `is_past` on the Event. So in that cases we just show the `event_date` the same way we show the rest of the data inside `p.event-metadata` block.

What do you think? For me even in the past events is good to show the data.

**Comment by duduribeiro** [comment]
`What do you think? For me even in the past events is good to show the data.`

for me too.

**Comment by duduribeiro** [comment]
And if not in the past, I show with that message you wrote?

**Comment by luanfonceca** [comment]
Yeah, and remove the "Due Date" message. As we will cover that two information with a single message, there is not need for it in there, but keep the color scheme in the whole message.

If it was not clear join us in the gitter.com/luanfonceca/speakerfight

**Comment by luanfonceca** [comment]
Hey @duduribeiro, can you please tell me how this is going?

**Comment by duduribeiro** [comment]
@luanfonceca updated. Can you check if the translation is right?

I wasn't able to test the translations locally (don't know why. I followed some blocktranslations on the project to guide me)

image

**Comment by luanfonceca** [comment]
Nice @duduribeiro, i will download it and run the translations after the merge.

File: deck/templates/event/snippets/event_list_item.html [hunk 1]
```diff
@@ -19,8 +19,25 @@ <h3 class="panel-title event-title">{{ event.title }}</h3>
...
+              {{ event.event_date }}
+            </span>
+          {% endif %}
         {% else %}
-          {% trans "Due date in" %} <span class="{% if event.due_date_is_close %}text-danger{% else %}text-warning{% endif %}">{{ event.due_date|timeuntil }}</span>
+          <span>
+            {% blocktrans with event_title=event.title %}
+              The event: <strong>{{ event_title }}</strong>
+            {% endblocktrans %}
+            {% if event.event_date %}
```
Comment (luanfonceca) [inline deck/templates/event/snippets/event_list_item.html]: I think that the `event_date` is required, isn't?

File: deck/forms.py [hunk 2]
```diff
@@ -30,6 +30,11 @@ class Meta:
                 'class': 'inline-input',
                 'placeholder': 'Due Date'
             }),
+            'event_date': CustomDateTimeWidget(attrs={
+                'id': 'id_event_date',
+                'class': 'inline-input',
+                'placeholder': 'Event Date'
```
Comment (luanfonceca) [inline deck/forms.py]: Can you please add the `gettext` in the  `'Event Date'` and  `'Due Date'`?

File: deck/templates/event/snippets/event_list_item.html (hunk 1 above)
Comment (duduribeiro) [reply deck/templates/event/snippets/event_list_item.html]: It is not. It is an optional field

File: deck/forms.py (hunk 2 above)
Comment (duduribeiro) [reply deck/forms.py]: sure

**Comment by duduribeiro** [comment]
@luanfonceca updated with `gettext`

**Comment by duduribeiro** [comment]
@luanfonceca is this right after the update with gettext?

**Pull request closed without merging**